cozy/cost_model.py

```diff
@@ -209,6 +280,80 @@ def worst_case_cardinality(e : Exp) -> DominantTerm:
         return DominantTerm.ONE
     return DominantTerm.N
 
+def _maintenance_cost(e : Exp, solver : ModelCachingSolver, op : Op, freebies : [Exp] = []):
+    """Determines the maintenance cost of an expression with regards to its
+    freebies and ops.
```

Comment:
A very small nitpick: "its" makes it sound like the freebies and ops are properties of the expression. They aren't. Rather, I view those as contextual clues for the cost model, which must answer "what is the cost of maintaining this expression in a world where these bits of state are free and these mutations might take place".